oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/FlatFileStore.java

```diff
@@ -38,12 +38,13 @@ public class FlatFileStore implements Iterable<NodeStateEntry>, Closeable {
     private final Closer closer = Closer.create();
     private final BlobStore blobStore;
     private final File storeFile;
+    private final File metadataFile;
     private final NodeStateEntryReader entryReader;
     private final Set<String> preferredPathElements;
     private final Compression algorithm;
     private long entryCount = -1;
 
-    public FlatFileStore(BlobStore blobStore, File storeFile, NodeStateEntryReader entryReader, Set<String> preferredPathElements, Compression algorithm) {
+    public FlatFileStore(BlobStore blobStore, File storeFile, File metadataFile, NodeStateEntryReader entryReader, Set<String> preferredPathElements, Compression algorithm) {
```

Comment:
Should we make an interface IndexingStore (like IncrementalStore) and make this class implement that. LAter on TreeStore can also implement the same interface.

Reply:
IndexStore is now used for both base and incrementalStore. As currently both stores have similar implementations. We can later add IncrementalStore in case implementation changes.